petastorm/fs_utils.py

```diff
@@ -134,8 +131,12 @@ def __init__(self, dataset_url, hadoop_configuration=None, connector=HdfsConnect
 
             storage_options = storage_options or {}
             protocol = self._parsed_dataset_url.scheme
-            self._filesystem = fsspec.filesystem(protocol, **storage_options)
-            self._filesystem_factory = lambda: fsspec.filesystem(protocol, **storage_options)
+            cls = fsspec.get_filesystem_class(protocol)
+            options = cls._get_kwargs_from_urls(self._dataset_url)
```

Comment:
Please don't use private methods of a third party library.

Reply:
All third party libraries being used by fsspec implement class AbstractFileSystem. AbstractFileSystem class have this method which either developer can over ride according to his/her needs. More info: https://filesystem-spec.readthedocs.io/en/latest/developer.html